index.js

```diff
@@ -133,7 +141,11 @@ function updateChildren (newNode, oldNode) {
 
       // Insert the node at the index if we couldn't morph or find a matching node
       } else {
-        oldNode.insertBefore(newChild, oldChild)
+        if (isProxy(newChild) && !newChild.isSameNode(oldChild) && newChild.realNode) {
+          oldNode.insertBefore(newChild.realNode, oldChild)
```

Comment:
We need some kind of check in here to make sure we aren't inserting a node in the wrong place.

Reply:
we could cheat and add a magic check to make sure we never insert proxy nodes perhaps?

Reply:
might create other fun (read: critical) failure scenarios tho, haha

Reply:
@yoshuawuyts yes. I had a version where it gave up every time it was about to insert a proxy, but it left things out of the DOM :/

so there are times when you do want to insert that realNode, just hard to know what the heuristics should be.